src/explore-education-statistics-admin/src/modules/find-statistics/components/KeyStatSelectForm.tsx

```diff
@@ -14,36 +17,35 @@ interface Props {
   label?: string;
 }
 
-const KeyIndicatorSelectForm = ({
+const KeyStatSelectForm = ({
   onSelect,
   onCancel = () => {},
   hideCancel = false,
-  label = 'Select a key indicator',
+  label = 'Select a key statistic',
 }: Props) => {
   const { availableDataBlocks } = useReleaseState();
   const [selectedDataBlockId, setSelectedDataBlockId] = useState('');
 
-  const [keyIndicatorDatablocks, setKeyIndicatorDatablocks] = useState<
-    DataBlock[]
-  >([]);
-
-  useEffect(() => {
-    setKeyIndicatorDatablocks(
-      availableDataBlocks.filter(db => {
-        return (
-          Object.entries(db.dataBlockRequest).filter(([key, val]) => {
-            // only want datablocks with (<2) of each filter type (filter, time period, location and indicator)
-            if (key === 'timePeriod') {
-              return !(
-                (val as TimePeriodQuery).startYear ===
-                (val as TimePeriodQuery).endYear
-              );
-            }
-            return Array.isArray(val) ? val.length > 1 : false;
-          }).length === 0
+  const keyStatDatablocks = useMemo(() => {
+    return availableDataBlocks.filter(db => {
+      const req = db.dataBlockRequest;
+      const timePeriod = req.timePeriod as TimePeriodQuery;
+      if (
+        Object.keys(req).filter(key => locationLevelKeys.includes(key as any))
```

Comment:
This isn't as efficient as the `some` check that we have later: `locationLevelKeys.some(key => req[key]?.length !== 1)`.

I would assign the result of the `some` as a variable like `hasSingleLocation` and then re-use it in the if.